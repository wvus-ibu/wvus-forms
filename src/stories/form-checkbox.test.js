import React from "react";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import { mount } from "enzyme";

import {
  propsChecked,
  propsUnchecked,
  propsDisabled,
  SimpleCheckboxForm
} from "./shared/checkbox-control-form";
import { WVUSForm } from "../index";

const Form = WVUSForm(SimpleCheckboxForm);

/**
 * SNAPSHOTS
 */

test("Snapshot: Checkbox Form renders correctly", () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snapshot: Checkbox renders checked state", () => {
  const tree = renderer
    .create(<SimpleCheckboxForm {...propsChecked} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snapshot: Checkbox renders unchecked state", () => {
  const tree = renderer
    .create(<SimpleCheckboxForm {...propsUnchecked} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snapshot: Checkbox can be disabled", () => {
  const tree = renderer
    .create(<SimpleCheckboxForm {...propsDisabled} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should be capable of having custom label and input classes", function() {
  const wrapper = mount(<SimpleCheckboxForm {...propsChecked} />);
  expect(wrapper.find("label").hasClass("opt-in-label")).toBe(true);
  expect(wrapper.find("input").hasClass("opt-in-input")).toBe(true);
});
