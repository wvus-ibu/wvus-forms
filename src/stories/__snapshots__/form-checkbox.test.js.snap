// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: Checkbox Form renders correctly 1`] = `
<div
  className=""
>
  <form>
    <div
      className="wvus-field-optIn wvus-field-newletter-optin checkbox-group"
    >
      <label
        className="opt-in-label checkbox-inline"
      >
        <input
          checked={true}
          className="opt-in-input"
          disabled={false}
          id="optIn"
          name="optIn"
          onChange={[Function]}
          type="checkbox"
        />
        Yes, I would like to receive the email newsletter.
      </label>
    </div>
  </form>
</div>
`;

exports[`Snapshot: Checkbox can be disabled 1`] = `
<form>
  <div
    className="wvus-field-optIn wvus-field-newletter-optin checkbox-group"
  >
    <label
      className="opt-in-label checkbox-inline"
    >
      <input
        checked={true}
        className="opt-in-input"
        disabled={false}
        id="optIn"
        name="optIn"
        onChange={[Function]}
        type="checkbox"
      />
      Yes, I would like to receive the email newsletter.
    </label>
  </div>
</form>
`;

exports[`Snapshot: Checkbox renders checked state 1`] = `
<form>
  <div
    className="wvus-field-optIn wvus-field-newletter-optin checkbox-group"
  >
    <label
      className="opt-in-label checkbox-inline"
    >
      <input
        checked={true}
        className="opt-in-input"
        disabled={false}
        id="optIn"
        name="optIn"
        onChange={[Function]}
        type="checkbox"
      />
      Yes, I would like to receive the email newsletter.
    </label>
  </div>
</form>
`;

exports[`Snapshot: Checkbox renders unchecked state 1`] = `
<form>
  <div
    className="wvus-field-optIn wvus-field-newletter-optin checkbox-group"
  >
    <label
      className="opt-in-label checkbox-inline"
    >
      <input
        checked=""
        className="opt-in-input"
        disabled={false}
        id="optIn"
        name="optIn"
        onChange={[Function]}
        type="checkbox"
      />
      Yes, I would like to receive the email newsletter.
    </label>
  </div>
</form>
`;
